R/MsExperiment.R

```diff
@@ -214,48 +214,99 @@ setMethod("readMsObject", signature(object = "MsExperiment",
 setMethod("readMsObject",
           signature(object = "MsExperiment",
                     param = "MetaboLightsParam"),
-          function(object, param, ...) {
-              url <- file.path(
-                  "https://ftp.ebi.ac.uk/pub/databases/metabolights/studies/public/",
-                  param@studyId, "/")
-              ## Retrieve the HTML content using curl system command
-              html_content <- system(paste("curl -s", url), intern = TRUE) ##idk if that works for all OS
-              if (all(grepl("The requested URL was not found on this server",
-                       html_content)))
-                  stop("Study not found. Please check the study ID and try again.")
-              files <- regmatches(html_content,
-                                  gregexpr("href=\"([^\"]+\\.txt|[^\"]+\\.tsv)\"",
-                                           html_content))
-              files <- unlist(files)
-              files <- gsub("href=\"|\"", "", files)
+          function(object, param, keepOntology = TRUE, keepProtocol = TRUE,
+                   simplify = TRUE, ...) {
+              if (!requireNamespace("MsBackendMetaboLights", quietly = TRUE)) {
+                  stop("Required package 'MsBackendMetaboLights' is missing. ",
+                       "Please install it and try again.", call. = FALSE)
+              }
+              pth <- MsBackendMetaboLights::mtbls_ftp_path(param@mtblsId)
+              all_fls <- MsBackendMetaboLights::mtbls_list_files(param@mtblsId)
 
-              ## check assay files
-              assays <- grep("a_", files, value = TRUE)
```

Comment:
I do this instead of using the `mtbls_list_files()` with a pattern, I benchmarked it and it is much faster in this context. Especially because then I would need to run it again later down for the sample info.